src/components/workProgress/WorkProgressInput.tsx

```diff
@@ -2,229 +2,160 @@ import { useMachine } from "@xstate/react";
-    };
-  }
+export const workProgressSearchSchema = (
+  workTypes: string[]
+): Yup.ObjectSchema => {
+  return Yup.object().shape({
+    searchType: Yup.string()
+      .oneOf(Object.values(WorkProgressSearchType))
+      .required(),
+    searchValues: Yup.array()
+      .of(Yup.string().required())
+      .when("searchType", {
+        is: (value: string) => value === WorkProgressSearchType.WorkType,
+        then: Yup.array().of(
+          workTypes
+            ? Yup.string().oneOf(workTypes).required()
+            : Yup.string().required()
+        ),
+      })
+      .when("searchType", {
+        is: (value: string) => value === WorkProgressSearchType.Status,
+        then: Yup.array().of(
+          Yup.string().oneOf(Object.values(WorkStatus)).required()
+        ),
+      }),
+  });
 };
-/**
- * Convert the data associated with the form to query input data structure.
- * @param workProgressInputFields
- */
-const formatFieldData = (
-  workProgressInputFields: WorkProgressInputData
-): WorkProgressQueryInput => {
-  const queryInput: WorkProgressQueryInput = {
-    workNumber: undefined,
-    workType: undefined,
-    status: undefined,
-  };
-  switch (workProgressInputFields.selectedType) {
-    case WorkProgressInputTypeField.WorkNumber: {
-      queryInput.workNumber = workProgressInputFields.selectedValue;
-      break;
-    }
-    case WorkProgressInputTypeField.WorkType: {
-      queryInput.workType = workProgressInputFields.selectedValue;
-      break;
-    }
-    case WorkProgressInputTypeField.Status: {
-      queryInput.status = workProgressInputFields.selectedValue as WorkStatus;
-      break;
-    }
-  }
-  return queryInput;
+
+type WorkProgressInputParams = {
+  urlParams: WorkProgressUrlParams;
+  workTypes: string[];
 };
 
 export default function WorkProgressInput({
-  initialValue,
-  onSubmitAction,
-}: {
-  initialValue: WorkProgressQueryInput;
-  onSubmitAction: (submitData: WorkProgressQueryInput) => void;
-}) {
-  //Initialize form data
-  const defaultInitialValues: WorkProgressInputData = {
-    types: Object.values(WorkProgressInputTypeField),
-    values: [],
-    selectedType: WorkProgressInputTypeField.WorkNumber,
-    selectedValue: "",
-    ...mergeFieldTypes(initialValue),
-  };
+  urlParams,
+  workTypes,
+}: WorkProgressInputParams) {
   const [current, send] = useMachine(
     createWorkProgressInputMachine({
-      workProgressInput: defaultInitialValues,
+      workProgressInput: urlParams,
     })
   );
 
   const {
-    workProgressInput: { types, values, selectedType, selectedValue },
-    serverError,
+    workProgressInput: { searchType, searchValues },
   } = current.context;
 
-  /**
-   * Form validation schema
-   */
-  const validationSchema: Yup.ObjectSchema = Yup.object().shape({
-    type: Yup.string().ensure(),
-    value: Yup.string().ensure(),
-  });
+  const generateValuesForType = React.useCallback(
+    (type: string): string[] => {
```

Comment:
Can this type be `WorkProgressSearchType` rather than a string?

Reply:
Done